### ISSUE TITLE
[BUG] Harvester v1.2.1 stuck on boot with BOSS-S1 hardware RAID

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#4784]
**Describe the bug**

Although we fixed the issue in v1.1.x, the customer still has an issue with 1.2.1.

**To Reproduce**
- Install Harvester v1.2.1.
- Reboot
- The OS hangs on "Welcome to GRUB!" for 30+ minutes.

**Expected behavior**

The OS should boot smoothly.

**Support bundle**

**Environment**
 - Harvester ISO version: v1.2.1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by bk201** [comment]
@megabreit Can you check if the issue description is still correct? And do you see the same issue in 1.2.0? Thanks.

**Comment by megabreit** [comment]
@bk201 We don't see this issue in 1.2.0.
The description is basically correct. It's still only happening with machines which contain the BOSS-S1 controller. There are some small differences.
- This time we see this error in the middle of the 30 mins:
```
error: ../../grub-core/fs/fshelp.c:258:file '/grub2/i386-pc/zfs.mod not found.
```
It's probably not related to the actual error, but maybe it helps to better track the problem.
- This time the boot counter is missing, so the boot does not automatically continue. The user needs to press Enter on the console.

The workaround `grub2-editenv /oem/grubcustom create` is still working, but maybe the actual cause is different this time.

**Comment by Vicente-Cheng** [comment]
> The workaround `grub2-editenv /oem/grubcustom create` is still working, but maybe the actual cause is different this time.

yap, that is what we are curious about. We are trying to reproduce it in our lab. And for sure I will backport this to v1.2.2.

**Comment by megabreit** [comment]
One, maybe important, detail: Our Harvester nodes are booting in BIOS mode. Not because of a technical decision, but just because it's default.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-9e663418-head` (24/01/12).

### Result
<details><summary>$\color{green}{\textsf{PASS}}$ On BOSS-S1 bare machine: Can install and reboot Harvester on BOSS-S1 RAID machine without stuck on boot $~~$</summary>

  1. Can install Harvester on BOSS-S1 card machine and without stuck on boot in GRUB
  
  https://github.com/harvester/harvester/assets/29251855/aff58090-097c-49d7-9887-eecc929a035d
  
  
  ![Screenshot from 2024-01-15 18-31-04](https://github.com/harvester/harvester/assets/29251855/b9901db9-7c2b-42f7-b7e6-bedb627b9711)

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ On BOSS-S1 bare machine: After installation, the `/oem/grubenv` and `/oem/grubcustom` files exists $~~$</summary>

  1. The `/oem/grubenv` and `/oem/grubcustom` files exists
  
  ```
  rancher@harvester-pxe-1:~> ls /oem/grubenv
  /oem/grubenv
  rancher@harvester-pxe-1:~> ls /oem/grubcustom
  /oem/grubcustom
  
  ```


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ On BOSS-S1 bare machine - After reboot: Harvester on BOSS-S1 RAID machine without stuck on boot $~~$</summary>

  1. Can reboot Harvester on BOSS-S1 card machine and without stuck on boot in GRUB
  

https://github.com/harvester/harvester/assets/29251855/24dff364-7e75-4039-bd72-b91e320e269e



 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ On local kvm: After installation, the `/oem/grubenv` and `/oem/grubcustom` files exists $~~$</summary>

  1. The `/oem/grubenv` and `/oem/grubcustom` files exists
  
  ```
  rancher@harv-240112:~> ls /oem/grubenv
  /oem/grubenv
  rancher@harv-240112:~> ls /oem/grubcustom
  /oem/grubcustom
  
  ```

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ On local kvm: After installation, Can install Harvester without stuck on boot $~~$</summary>

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ On local kvm: After reboot, Can reboot Harvester  without stuck on boot $~~$</summary>

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ On local kvm: After reboot, the `/oem/grubenv` and `/oem/grubcustom` files exists $~~$</summary>

  1. The `/oem/grubenv` and `/oem/grubcustom` files exists
  
  ```
  rancher@harv-240112:~> ls /oem/grubenv
  /oem/grubenv
  rancher@harv-240112:~> ls /oem/grubcustom
  /oem/grubcustom
  
  ```

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ On ipxe installation: After installation, the `/oem/grubenv` and `/oem/grubcustom` files exists $~~$</summary>

  1. The `/oem/grubenv` and `/oem/grubcustom` files exists
  
  ```
  rancher@harvester-node-0:~> ls /oem/grubenv
  /oem/grubenv
  rancher@harvester-node-0:~> ls /oem/grubcustom
  /oem/grubcustom
  ```

 </details>


### Test Information
* Test Environment: Single nodes harvester on equinix bare machines and local kvm machine
* Harvester version: `master-9e663418-head` (24/01/12)

### Verify Steps
<details><summary>On local kvm machine </summary>

  1. ISO Install Harvester v1.3.0 master release on BIOS kvm 
  2. Check no stuck during the installation process
  3. Check the check the `/oem/grubenv` and `/oem/grubcustom` files exists
  4. Reboot the machine, check no stuck 
  5. Check no stuck, can smoothly boot up Harvester

</details>

<details><summary>On BOSS-S1 raid baremetal machine</summary>

1. Ensure the machine boot in BIOS mode
  ![image](https://github.com/harvester/harvester/assets/29251855/55532b33-9efe-464f-97f0-3bd25e0a2b67)

2. Ensure the machine have BOSS S1 card
  ![image](https://github.com/harvester/harvester/assets/29251855/28e95bf1-b896-419c-a7a9-ffd7e9f8e592)

3. Use terraform equinix to ipxe install Harvester v1.3.0 master release on BIOS kvm 
4. Check no stuck during the installation process
5. Check the `/oem/grubenv` and `/oem/grubcustom` files exists
6. Reboot the machine, check no stuck 
7. Check the  `/oem/grubenv` and `/oem/grubcustom` files exists after reboot machine

</details>
